packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -377,11 +380,39 @@ class GA4 extends integrations.Integration<GA4IntegrationOptions> {
       this.processInteractionEvents(commandList, data?.event);
 
       each(commandList, command => {
+        this.attachUniqueEventIdToCommand(command, data);
+
         window.gtag.apply(null, command);
       });
     }
   }
 
+  /**
+   * Attaches the unique event ID generated by the analytics package.
+   *
+   * @param command - The command being processed.
+   * @param data - Event data provided by analytics.
+   *
+   * @returns The modified command.
+   */
+  attachUniqueEventIdToCommand(
+    command: GA4Command,
+    data:
+      | TrackEventData
+      | PageviewEventData
+      | LoadIntegrationEventData
+      | SetUserEventData,
+  ) {
+    const eventProperties = command[2] as Record<string, string>;
+    const uniqueEventId = get(data, 'context.event.__uniqueEventId');
```

Comment:
Also, we could extract the `__uniqueEventId` to a constant and replace it in all integrations. That way we can easily change the name of the `__uniqueEventId` property in a single point.